### PR TITLE
Updates to enable material property checks on boundary

### LINKED ISSUE #5309: Retrieving a material property in a BC does not trigger an error if it's not supplied
Reported by Alex Lindsay. 

@dschwen made the following comment:
This can be replicated by removing the material k3bnd in
moose/test/tests/materials/boundary_material/bnd_coupling_vol.i
…

### LINKED ISSUE #9482: getMaterialProperty in DGKernel is not checked for integrity
### Description of the enhancement or error report
Looks like `DGKernel::getMaterialProperty` is not mark the consumer of the material property, thus the problem integrity check is not covering the case where `DGKernel` is trying to use a un-declared property.

### Rationale for the enhancement or information for reproducing the error
DG kernels may not properly set up.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
Input robustness.

### LINKED ISSUE #9835: SideIntegralVariablePostprocessor doesn't throw an error if getting a material that doesn't exist
### Description of the enhancement or error report
So, I just discovered that if a material property requested by a postprocessor based on SideIntegralVariablePostprocessor doesn't exist, the postprocessor will happily run producing junk output rather than throw an error.

### Rationale for the enhancement or information for reproducing the error
This behaviour should produce an error

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
Make this postprocessor more robust

### PR DESCRIPTION
**Pull request by aeslaughter**
This enables boundary material property checks and cleans up the constructor of the (Two)MaterialPropretyInterface by utilizing the MooseObject pointer.

(closes #9835)
(closes #5309)
(closes #9482)

### CODE REVIEW

**Comment by aeslaughter** [comment]
@andrsd This might interest you: https://www.moosebuild.org/job/116300/

**Comment by aeslaughter** [comment]
@joshuahansel Can you take a look at this failure:  https://www.moosebuild.org/job/116300/. I am adding improved boundary material checks and this happened, seems like it could be important.

**Comment by joshuahansel** [comment]
@aeslaughter CIVET claims that I do not have permission to view the results. I view RELAP-7 results often with my Gitlab account, but maybe I've never had to view results on moosebuilds with my github account... Anyway, it appears I'll need some additional permissions.

**Comment by joshuahansel** [comment]
@aeslaughter Ok, I believe I see the problem. What I don't understand yet is why this isn't already broken in RELAP-7 - I would think that this error should be present even without your changes. I'll need to investigate this.

**Comment by joshuahansel** [comment]
Never mind my previous comment. The issue was that before these changes, a garbage boundary property was being created due to a non-existent material property. The breakage of the 2 tests with these changes is expected and will be fixed at a later date.

**Comment by aeslaughter** [comment]
@permcody This is ready, the RELAP7 failures will be handled by @joshuahansel at a later date.

File: test/tests/materials/boundary_material/tests [hunk 1]
```diff
@@ -12,4 +12,4 @@
     input = 'elem_aux_bc_on_bnd.i'
     exodiff = 'elem_aux_bc_on_bnd_out.e'
   [../]
-[]
+s[]
```
Comment (permcody) [inline test/tests/materials/boundary_material/tests]: What's this little guy?

File: test/tests/misc/block_boundary_material_check/bc_check.i [hunk 2]
```diff
@@ -0,0 +1,31 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 1
+[]
+
+[Variables]
+  [u]
```
Comment (permcody) [inline test/tests/misc/block_boundary_material_check/bc_check.i]: Are we sure we want to drop the dots and slashes? I really think we should wait for the python support to catch up. How does this work with Peacock?

**Review by permcody — CHANGES_REQUESTED** [review]
Stray character. Also, I recommend that we hold off on the new HIT syntax for just a little while longer. @rwcarlsen is currently working on some python solution.

File: test/tests/misc/block_boundary_material_check/bc_check.i (hunk 2 above)
Comment (aeslaughter) [reply test/tests/misc/block_boundary_material_check/bc_check.i]: I will fix these up.

**Review by permcody — APPROVED** [review]

**Comment by lindsayad** [comment]
@aeslaughter Thanks for this work!

**Comment by cpgr** [comment]
Thanks @aeslaughter!

**Pull request merged**